src/ewatercycle/_forcings/caravan.py

```diff
@@ -0,0 +1,240 @@
+import xarray as xr
+import numpy as np
+import pandas as pd
+from cartopy.io import shapereader
+import fiona
+
+from pathlib import Path
+from typing import Type
+
+import requests
+from zipp import zipfile
+
+from ewatercycle.base.forcing import DefaultForcing, LumpedUserForcing
+from ewatercycle.util import get_time
+
+COMMON_URL = "ca13056c-c347-4a27-b320-930c2a4dd207"
+OPENDAP_URL = f"https://opendap.4tu.nl/thredds/dodsC/data2/djht/{COMMON_URL}/1/"
+SHAPEFILE_URL = f"https://data.4tu.nl/file/{COMMON_URL}/bbe94526-cf1a-4b96-8155-244f20094719"
+
+PROPERTY_VARS = ['timezone',
+                 'name',
+                 'country',
+                 'lat',
+                 'lon',
+                 'area',
+                 'p_mean',
+                 'pet_mean',
+                 'aridity',
+                 'frac_snow',
+                 'moisture_index',
+                 'seasonality',
+                 'high_prec_freq',
+                 'high_prec_dur',
+                 'low_prec_freq',
+                 'low_prec_dur']
+
+RENAME_ERA5 = {'total_precipitation_sum':'pr',
+                'potential_evaporation_sum':'evspsblpot',
+                 'temperature_2m_mean': 'tas',
+                 'temperature_2m_min': 'tasmin',
+                 'temperature_2m_max': 'tasmax',
+                 'streamflow':'Q'}
+class Caravan(DefaultForcing):
+    """Retrieves specified part of the caravan dataset from the OpenDAP server.
+
+    Examples:
+
+    The caravan dataset is an already prepared set by Frederik Kratzert,
+    (see https://doi.org/10.1038/s41597-023-01975-w).
+
+    This retrieves it from the OpenDAP server of 4TU,
+    (see https://doi.org/10.4121/bf0eaf7c-f2fa-46f6-b8cd-77ad939dd350.v4).
+
+    This can be done by specifying the
+
+    .. code-block:: python
+
+        from pathlib import Path
+        from rich import print
+        from ewatercycle.forcing import sources
+
+        path = Path.cwd()
+        forcing_path = path / "Forcing"
+        experiment_start_date = "1997-08-01T00:00:00Z"
+        experiment_end_date = "2005-09-01T00:00:00Z"
+        HRU_id = 1022500
+
+        camels_forcing = sources['LumpedCaravanForcing'].retrieve(start_time = experiment_start_date,
+                                                                  end_time = experiment_end_date,
+                                                                  directory = forcing_path / "Camels",
+                                                                  basin_id = f"camels_0{HRU_id}"
+                                                                )
+        which gives somthing like:
+
+        .. code-block:: python
+
+            LumpedCaravanForcing(
```

Comment:
It looks a lot like the other forcings now, which is nice. That should make it easier to swap this forcing with, e.g. Makkink, or the other way around.

Reply:
Yup, might also be nice to also have the seperate Camels data sets as these do vary slightly. (But thats more work for another time)